src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,304 @@
+                bye();
+                break;
+            case "HELP":
+                printHelpInfo();
+                break;
+            default:
+                printErrorInfo();
+                break;
+            }
+
+            autoSave(autoSaveFile);
+        }
+    }
+
+    private static void processInput(Scanner in) {
+            userInputLine = in.nextLine();
+            command = processCommand();
+    }
+
+    private static void printErrorInfo() {
+        System.out.println(HORIZONTAL_LINE + "\n"
+                + "☹ OOPS!!! I'm sorry, but I don't know what that means :-(\n"
+                + " pls refer to the commands by typing 'help'.\n"
+                + HORIZONTAL_LINE );
+    }
+    private static void printHelpInfo() {
+        System.out.println(HORIZONTAL_LINE + "\n"
+                + "LIST, DONE, TODO, DEADLINE, EVENT, BYE, HELP\n"
+                + HORIZONTAL_LINE);
+    }
+
+    public static void showLogo() {
+        String logo = " ____        _\n"
+                    + "|  _ \\ _   _| | _____\n"
+                    + "| | | | | | | |/ / _ \\\n"
+                    + "| |_| | |_| |   <  __/\n"
+                    + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println(HORIZONTAL_LINE+ "\n" + logo);
+    }
+    public static void greet() {
+        String greeting = "Hello! I'm duke.Duke\n"
+                        + "What can I do for you?\n";
+        System.out.println(HORIZONTAL_LINE + "\n" + greeting + HORIZONTAL_LINE);
+    }
+    public static void bye() {
+        String bye_word = "Bye. Hope to see you again soon!";
+        System.out.println(HORIZONTAL_LINE + "\n" + bye_word + "\n" + HORIZONTAL_LINE);
+    }
+
+    public static void addTask(Task t) {
+        tasks.add(t);
+        taskCount = tasks.size();
+        System.out.println(HORIZONTAL_LINE + "\n"
+                + "Got it. I've added this task:\n"
+                + "  " +t.toString() + "\n"
+                + "Now you have "+ taskCount + " duke.tasks in the list\n"
+                + HORIZONTAL_LINE);
+
+    }
+    public static void listTasks() {
+        System.out.println(HORIZONTAL_LINE + "\n"
+                + "Here are the duke.tasks in your list:");
+        for (int i = 0; i< taskCount; i++) {
+            System.out.println((i+1)+"."+tasks.get(i).toString());
+        }
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    public static Boolean checkTaskNum() {
+        boolean isNumValid = true;
+        try{
+            int inputNum = Integer.parseInt(description);
+            if (inputNum > taskCount || taskCount == 0 || inputNum == 0) {
+                throw new DukeException();
+            }
+        }catch (NumberFormatException e) {
+            isNumValid = false;
+            System.out.println(HORIZONTAL_LINE+"\n"
+                    + "☹ OOPS!!! The task number of the one you have done is invalid!\n"
+                    +HORIZONTAL_LINE);
+        }catch (DukeException e) {
```

Comment:
This catch block is unnecessary, you can just add the logic in the block to the if block above instead of throwing a error there. You should throw error if you want a higher level method to handle it.